src/simulations/collect.jl

```diff
@@ -136,6 +138,7 @@ function run!(
     end
 
     s = 0
+    p = showprogress ? typeof(n) <: Int ? ProgressMeter.Progress(n, 1) : [0] : nothing
```

Comment:
This is not the correct way to use the `ProgressMeter` package. Instead you need to initialize a proper progress meter as indicated by the package's API. `p = ProgressMeter.Progress(n, 0, "Performing `Agents.run!`...")` and later  down the line call `next!(p)`.

Reply:
Furthermore, the ProgressMeter.jl package has a `ProgressUnknown` type that we can use when `n` is a function: https://github.com/timholy/ProgressMeter.jl#progress-meters-for-tasks-with-an-unknown-number-of-steps

Reply:
Thanks for your feedback. I wasn't aware of `ProgressUnknown`.